students/IanMayther/mailroom/Mailroom.py

```diff
@@ -62,68 +96,68 @@ def email(TO, gift_amount):
 \n
 Sincerely,\n
 Derek Zoolander\n
-Founder and C.E.O. of Derek Zoolander Charity for Ants Who Can't Read Good (DZCAWCRG)"""
+Founder and C.E.O. of Derek Zoolander Charity for Ants Who Can't Read Good (DZCAWCRG)\n"""
     return body
 
 
 #Create Report
-def print_report(my_List = []):
+def print_report(my_List = {}):
     #Header
+    print("\n")
     print("{0:<25s}|{1:^15s}|{2:^15s}|{3:>12s}".format("Donor Name", "Total Given", "# of Gifts","Avg. Gift"))
     print("-" * 72)
+    
     #Table Data
-    for i in range(len(my_List)):
-        #Process Data
-        #Donor
-        col = my_List[i][1]
-        #Sum Total 
-        total = sum(col)
+    for k, v in sorted(donors.items(), key=lambda t: sum(t[1]), reverse=True):
+        #Process Data 
+        total = sum(v)
         #Gift Count  
-        No_Gifts = len(my_List[i][1])
+        No_Gifts = len(v)
         #Calc Average
         Ave_Gift = total / No_Gifts
         #Print Table
-        print("{0:<25s}${1:>14.2f}{2:>17d}  ${3:>11.2f}".format(my_List[i][0],total, No_Gifts, Ave_Gift, end =''))    
+        print("{0:<25s}${1:>14.2f}{2:>17d}  ${3:>11.2f}".format(k, total, No_Gifts, Ave_Gift, end =''))
+    print("\n") 
     return
 
+#Send Letter
+def send_letter():
+    path = pathlib.Path.cwd()
+    for k, v in donors.items():
+        file_name = k + '_Thank you Letter.txt'
+        with open(file_name, 'w') as l:
+            l.write(email(k, sum(v)))
+    print(f"Sending Letters to disk: {path}\n")
+    pass
+
+
+
+#Quit
+def quit():
+    print("Quitting, Thank you.")
+    return "quit"
+
+#Main Menu Options
+def main_menu(prompt, dict_choice):
+    while True:
+        choice = input(prompt)
+        if dict_choice[choice]() == "quit":
+            break
+
+choice_menu = ("Choose an Action:\n"
+            "\n"
+            "1 - Send Thank You to Single Donor.\n"
+            "2 - Create Report.\n"
+            "3 - Send Letters to ALL Donor.\n"
+            "4 - Quit.\n")
+
+main_selections = {"1" : receiver,
+                    "2" : print_report,
+                    "3" : send_letter,
+                    "4" : quit,
+                    }
+    
+
```

Comment:
Looks good but a couple of functional issues.  Unable to 'list' all donors and when selecting an existing donor, unable to input a donation amount.  Please take another look and let me know if you need help.